src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp

```diff
@@ -692,3 +729,54 @@ void GraphingCalculator::OnVisualStateChanged(Object ^ sender, VisualStateChange
 {
     TraceLogger::GetInstance()->LogVisualStateChanged(ViewMode::Graphing, e->NewState->Name, false);
 }
+
+void GraphingCalculator::OnColorValuesChanged(Windows::UI::ViewManagement::UISettings ^ sender, Platform::Object ^ args)
+{
+    WeakReference weakThis(this);
+    this->Dispatcher->RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, ref new Windows::UI::Core::DispatchedHandler([weakThis]() {
+                                   auto refThis = weakThis.Resolve<GraphingCalculator>();
+                                   if (refThis != nullptr && refThis->IsMatchAppTheme)
+                                   {
+                                       refThis->UpdateGraphTheme(Application::Current->RequestedTheme);
+                                   }
+                               }));
+}
+
+void GraphingCalculator::UpdateGraphTheme(ApplicationTheme appTheme)
+{
+    if (m_accessibilitySettings->HighContrast)
+    {
+        VisualStateManager::GoToState(this, L"GrapherHighContrast", true);
+        return;
+    }
+    if (appTheme == ApplicationTheme::Dark)
```

Comment:
Nit: `else if` or `switch`

Reply:
Instead of calculating appTheme outside this method, for no reasons sometimes (high contrast), we should probably remove the argument and directly calculate the value to use in the function.